File: javaFiles/Abstraction.java
package javaFiles;

public abstract class Abstraction {
    abstract void calculateArea();
    abstract void calculateCircumference();
}

class Square extends Abstraction {
    private double side;

    public Square(double side) {
        this.side = side;
    }

    @Override
    void calculateArea() {
        double area = side * side;
        System.out.println("Area of Square: " + area);
    }

    @Override
    void calculateCircumference() {
        double circumference = 4 * side;
        System.out.println("Circumference of Square: " + circumference);
    }
}

class AbstractionDemo {
    public static void main(String[] args) {
        Abstraction square = new Square(5);
        square.calculateArea();
        square.calculateCircumference();
    }
}

File: dartFiles/Abstraction.dart
abstract class Shape {
  void calculateArea();
  void calculateCircumference();
}

class Square extends Shape {
  double? _side;

  Square(double side) {
    this._side = side;
  }

  @override
  void calculateArea() {
    double area = _side! * _side!;
    print("Area of Square: $area");
  }

  @override
  void calculateCircumference() {
    double circumference = 4 * _side!;
    print("Circumference of Square: $circumference");
  }
}

void main() {
  Shape square = Square(5);
  square.calculateArea();
  square.calculateCircumference();
}